sdks/cpp/common/include/ParamAccessor.h

```diff
@@ -666,6 +666,20 @@ class ParamAccessor {
     }
 
   private:
+     /**
+     * @brief checks if the src value is the correct type to set this parameter
+     * @param src the value to compare against
+     * @param idx the index into the array
+     * @return true if the src value is the correct type to set this parameter
+     * 
+     * for arrays if idx is not kParamEnd, the src value must be proper type for the array
+     * 
+     * @todo check struct array and variant array kinds
+     * @todo check that structs contain same sub-params
+     * @todo determine how array size should be compared when idx is kParamEnd
+     */
+    bool sameKind(const Value& src, const ParamIndex& idx) const;
```

Comment:
because ParamIndex is just an int32, it should be passed by value. In general, only pass by reference objects that are bigger than a single word - e.g. strings, vectors, class instances, ...